### ISSUE TITLE
Upgrade the `Dockerfile` to utilize buildkit's directory caching for much faster builds

### ISSUE THREAD

**Issue by spkane** [asciidoctor/docker-asciidoctor#413]
This makes it possible to cache dep downloads from tools like `apt,` `go get,` `gem,` etc.

See: https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/reference.md#run---mounttypecache

**Comment by dduportal** [comment]
Good idea. Have you already used it in CI environments (i’m wondering how it would work with GitHub Actions ephemeral workers)

**Comment by spkane** [comment]
It would likely take some work to get it working well in the CI/CD pipeline. It would likely require either taking advantage of Github caching (https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflow) or getting Docker to store the cache somewhere remote, or some combination of the two.

That being said, it would improve the local workflow, no matter what. :-)

**Comment by Banane9** [comment]
Was there any progress on this, or does the slow behavior I'm experiencing have another cause?

I've been using the (latest) docker image to build about 90kB worth of documentation that generates a ~100 page PDF document. This seems to take around 2 minutes each time when done through a docker run command, which is a far cry from the 100kB in one second claim on the Asciidoctor Docs page :(

**Comment by dduportal** [comment]
> Was there any progress on this, or does the slow behavior I'm experiencing have another cause?
> 
> I've been using the (latest) docker image to build about 90kB worth of documentation that generates a ~100 page PDF document. This seems to take around 2 minutes each time when done through a docker run command, which is a far cry from the 100kB in one second claim on the Asciidoctor Docs page :(

This is unrelated :) But it could be worth an issue: Do you mind opening a new one with more details?

**Comment by Banane9** [comment]
> This is unrelated :) But it could be worth an issue: Do you mind opening a new one with more details?

Sure, what details would be useful for you?

**Comment by dduportal** [comment]
> > This is unrelated :) But it could be worth an issue: Do you mind opening a new one with more details?
> 
> Sure, what details would be useful for you?

The usual when you report an issue: environment, tools and their versions, reproduction case ideally, etc.
